cmd/sriov-network-config-daemon/service.go

```diff
@@ -0,0 +1,166 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	sriovv1 "github.com/k8snetworkplumbingwg/sriov-network-operator/api/v1"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/host"
+	plugin "github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/plugins"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/plugins/generic"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/systemd"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/utils"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/version"
+)
+
+var (
+	serviceCmd = &cobra.Command{
+		Use:   "service",
+		Short: "Starts SR-IOV service Config",
+		Long:  "",
+		Run:   runServiceCmd,
```

Comment:
Maybe here it's better to use `cobra.Command.RunE` 

```
$ export DOCKERFILE=Dockerfile.sriov-network-config-daemon                                                                                                                                                                                             
$ export APP_NAME=sriov-network-config-daemon
$ make image
...
$ docker run -it ghcr.io/k8snetworkplumbingwg/sriov-network-config-daemon /usr/bin/sriov-network-config-daemon service 
I1206 11:00:24.618527       1 service.go:49] Starting sriov-config-service
E1206 11:00:24.618655       1 service.go:52] failed to read list of supported nic ids
E1206 11:00:24.618677       1 systemd.go:136] WriteSriovResult(): failed to create sriov result file on path /etc/sriov-operator/sriov-interface-result.yaml: open /etc/sriov-operator/sriov-interface-result.yaml: no such file or directory
E1206 11:00:24.618685       1 service.go:59] sriov-config-service failed to write sriov result file with content {Failed failed to read list of supported nic ids: stat /etc/sriov-operator/sriov-supported-nics-ids.yaml: no such file or directory} error: open /etc/sriov-operator/sriov-interface-result.yaml: no such file or directory
$ echo $?
0
```

Otherwise, systemd unit will be marked as completed with success, even in case of error.

Reply:
right thanks!